libbeat/idxmgmt/idxmgmt.go

```diff
@@ -57,25 +51,38 @@ type Supporter interface {
 
 	// Manager creates a new manager that can be used to execute the required steps
 	// for initializing an index, ILM policies, and write aliases.
-	Manager(client ESClient, assets Asseter) Manager
+	Manager(client ClientHandler, assets Asseter) Manager
 }
 
 // Asseter provides access to beats assets required to load the template.
 type Asseter interface {
 	Fields(name string) []byte
 }
 
-// ESClient defines the minimal interface required for the index manager to
-// prepare an index.
-type ESClient interface {
-	Request(method, path string, pipeline string, params map[string]string, body interface{}) (int, []byte, error)
-	GetVersion() common.Version
-}
-
 // Manager is used to initialize indices, ILM policies, and aliases within the
 // Elastic Stack.
 type Manager interface {
-	Setup(forceTemplate, forcePolicy bool) error
+	Setup(template, ilm LoadMode) error
+}
+
+// LoadMode defines the mode to be used for loading idxmgmt related information.
+// It will be used in combination with idxmgmt configuration settings.
+type LoadMode uint8
+
+const (
+	// LoadModeUnset indicates that no specific mode is set, loading should be derived only from config.
```

Comment:
Good comment, but: What is the expected default behavior if we derive the mode from the configs? LoadModeEnabled? In libbeat we prefer same default behavior for common settings like enabled/overwrite.

Reply:
The `mode` itself is set according to the config or the respective default values. I cannot document here which default this is, as it depends on the information the mode is used with. 
E.g. for `template` the config default is `enabled:true`, so when used in combination, it would be `LoadModeEnabled`.
For `ilm` default is also set to `true` in libbeat, but might be different in other beats. 
I'll try to make this more clear in the comment though.